Classes/Search/GithubClient+Search.swift

```diff
@@ -7,19 +7,26 @@
 //
 
 import Foundation
+import Apollo
 
 extension GithubClient {
     
     enum SearchResultType {
         case error
         case success(String?, [SearchRepoResult])
     }
-    
-    func search(query: String, before: String? = nil, containerWidth: CGFloat, completion: @escaping (SearchResultType) -> ()) {
+
+    @discardableResult
+    func search(
+        query: String,
+        before: String? = nil,
+        containerWidth: CGFloat,
+        completion: @escaping (SearchResultType) -> ()
+        ) -> Cancellable {
         let query = SearchReposQuery(search: query, before: before)
-        
-        fetch(query: query) { (result, error) in
-            guard error == nil, result?.errors == nil else {
+
+        return fetch(query: query) { (result, error) in
+            guard (error == nil || (error! as NSError).code == -999), result?.errors == nil else {
```

Comment:
Could we refactor this slightly to drop the force unwrap? (I get that it's safe because of the nil check before, but would prefer not to have it if we can easily avoid it!)

Reply:
@Sherlouk 
How do you feel about global-scope private functions? 

```swift
extension GitHubClient { 
  ...
  guard errorIsNilOrCancelled(error), ...
  ...
}

private func errorIsNilOrCancelled(_ error: Error?) -> Bool {
    guard let error = error else { return true }
    return (error as NSError).code == -999
}
```

Reply:
I don't terribly mind this, alternatively just using optionals: `guard (error == nil || (error as? NSError)?.code == -999), result?.errors == nil else {` should have the same effect (I think)?

Leaving it up to you, as long as we have no pesky exclamation marks 😉

Reply:
@Sherlouk I'd love to use an optional here but `Error?` doesn't bridge to `NSError` -- the compiler asks that we use a hard-cast `as!` for that conversion